src/frontend/src/app/components/events/event-add/event-add.component.ts

```diff
@@ -49,5 +59,10 @@ export class EventAddComponent implements OnInit {
     return this.eventForm.get(groupName).get(controlName).hasError('required') ? 'You must enter a value' :
            '';
   }
+
+  comparer(o1: Tag, o2: Tag): boolean {
+    // if possible compare by object's name property - and not by reference.
+    return o1 && o2 ? o1.name === o2.name : o2 === o2;
```

Comment:
I'm not the greatest with Typescript and @jfranco1701 is a way better reference for this but this line feels... odd. What if `o1` is provided and not `o2`, would the `false` part fail? Could we shrink this function to just `return o1.name === o2.name` by using parameter defaults?

Reply:
If one is null the false part returns false ( `o2 === o2` ). Funny thing is that if both are null, it returns true. I found this workaround on github somewhere because the list of available tags are different than the tags on the 'Event' object.